### ISSUE TITLE
RUF009 triggers for intentional constant computation

### ISSUE THREAD

**Issue by RonnyPfannschmidt** [astral-sh/ruff#10069]
https://play.ruff.rs/aa67fe33-c09c-4a7f-91fc-5fe9b5b98425

in pytest i replaced a magic constant with its more comprehensible computation

```
# minimized reducers
import datetime
import dataclasses

@dataclasses.dataclass
class MockTiming:
   
    _current_time: float = datetime(2020, 5, 22, 14, 20, 50).timestamp()
    _current_time_before: float = 1590150050.0
```

i recon that its possibly hard to sanely filter false positives
i do wonder if given type annotation awareness it could permit in case the value is a builtin immutable

**Comment by MichaReiser** [comment]
Yeah, it might be possible but it isn't sufficient to only test if the result type is immutable, ruff would also need to know if the function is idempotent (does not depend on any global or system state) as well as all arguments passed to the call. 

I understand that the result here is undesired but getting this right requires more analysis than ruff is capable today. I recommend manually suppressing the rule for this case and documenting why it is safe (documenting this might also help future readers).
